tests/test_iterative.py

```diff
@@ -0,0 +1,50 @@
+from pytest import raises
+from pydecorators.iterative import loop,retry,average
+
+
+def test_loop_n():
+    @loop(10)
+    def add(a,b):
+        return a+b
+    assert add(1,1) == [2] * 10
+
+def test_loop_single_default():
+    @loop
+    def add(a,b):
+        return a+b
+    assert add(1,1) == [2] * 5
+
+def test_retry_3():
+    tries = 0 
+    @retry(3)
+    def t():
+        nonlocal tries
+        tries+=1
+        if tries != 3:
+            raise Exception
+        return None
+
+    assert t() == None and tries == 3
```

Comment:
make it in two different assert, to easier see what's wrong

Reply:
and you can change from None to some different value, to better see if result is correct